android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/ResourceExtension.kt

```diff
@@ -352,6 +352,19 @@ fun String.resourceClassType(): Class<out Resource> =
  */
 fun String.extractLogicalIdUuid() = this.substringAfter("/").substringBefore("/")
 
+/**
+ * A function that extracts only the Resource name part of a resource logicalId.
+ *
+ * Examples:
+ * 1. "Group/0acda8c9-3fa3-40ae-abcd-7d1fba7098b4/_history/2" returns
+ *
+ * ```
+ *    "Group".
+ * ```
+ * 2. "Group/0acda8c9-3fa3-40ae-abcd-7d1fba7098b4" returns "Group".
+ */
+fun String.extractResourceTypeName() = this.substringBefore("/")
```

Comment:
Is this String extension really necessary? I think we can just call the `substringBefore("/")` directly unless this function is called multiple times and this makes sense. Since the extension is on String class it shouldn't be in `ResourceExtension` file

Reply:
As of now it's only called in this location